src/libutil/typedesc.cpp

```diff
@@ -175,8 +175,6 @@ TypeDesc::c_str () const
     std::string result;
     if (aggregate == SCALAR)
         result = basetype_name[basetype];
-    else if (aggregate == MATRIX44 && basetype == FLOAT)
-        result = "matrix";
```

Comment:
OIIO is a dependency for OSL, not the other way around, so a few classes that are needed for both live in OIIO but may have some particular design quirks that came directly from imagining how it would be used in a shading system or renderer.

For c_str(), I would like to make a pitch for keeping "matrix" to mean the 'usual' case of 4x4 transformation matrices used in computer graphics, and only need the numbers for other cases (3x3 is the only other one we have planned at the moment). I think that's similar in spirit to how "vector" is understood to be the usual case of vector3, but we use "vector2" or "vector4"